ansible/roles/schulcloud-server-tspsync/tasks/main.yml

```
@@ -111,3 +111,24 @@
       state: "{{ 'present' if WITH_TSP_SYNC|bool else 'absent'}}"
     tags:
       - configmap
+
+  - name: remove old API TSP Sync Init Job
+    kubernetes.core.k8s:
+      kubeconfig: ~/.kube/config
+      namespace: "{{ NAMESPACE }}"
+      api_version: batch/v1
+      kind: Job
+      name: api-tsp-sync-init
+      state: absent
+      wait: yes
+    tags:
+      - job
```

Missing when condition for removig it
```suggestion
    when: not WITH_TSP_SYNC
    tags:
      - job
```

---

Added